src/Cool/DidYouKnow/didYouKnowModal.cpp

```diff
@@ -0,0 +1,98 @@
+#include "didYouKnowModal.hpp"
+#include <imgui.h>
+#include <vcruntime.h>
+#include <string>
+
+void DidYouKnowModal::open()
+{
+    auto difference = std::chrono::system_clock::to_time_t(std::chrono::system_clock::now()) - std::chrono::system_clock::to_time_t(_current_timestamp);
+
+    size_t time_to_wait = 1 * 60 * 60; // 1 hour
+
+    if (!_has_been_opened && difference > time_to_wait)
+    {
+        _has_been_opened = true;
+        ImGui::OpenPopup(_id.c_str());
+        _current_timestamp = std::chrono::system_clock::now();
+        return;
+    }
+}
+
+void DidYouKnowModal::prepare_next_tip()
+{
+    _current_tip_index++;
+    if (_current_tip_index >= _all_tips.size())
+        _current_tip_index = 0;
+    _text = _all_tips[_current_tip_index];
+}
+
+bool DidYouKnowModal::is_open() const
+{
+    return ImGui::BeginPopupModal(_id.c_str(), NULL, ImGuiWindowFlags_AlwaysAutoResize);
+}
+
+void DidYouKnowModal::display_modal_content()
+{
+    ImGui::Text(_text.c_str());
+    ImGui::Separator();
+
+    if (ImGui::Button("OK", ImVec2(120, 0)))
+    {
+        ImGui::CloseCurrentPopup();
+        prepare_next_tip();
+    }
+
+    ImGui::SameLine();
+    ImGui::SetItemDefaultFocus();
```

Comment:
Why is this line here ? If there is a good reason, we need a comment explaining it because it is unusual to do this.